cmd/drand-cli/cli_test.go

```diff
@@ -664,6 +664,64 @@ func getSBFolderStructure() string {
 	return tmp
 }
 
+func TestDrandReloadBeacon(t *testing.T) {
+	sch := scheme.GetSchemeFromEnv()
+	beaconID := common.GetBeaconIDFromEnv()
+
+	// beacon id need to have a value in order to stop one beacon process
+	// if beacon id is empty, the id will be "default" internally
+	if beaconID == "" {
+		beaconID = common.DefaultBeaconID
+	}
+
+	n := 4
+	instances, tempPath := launchDrandInstances(t, n)
+	defer os.RemoveAll(tempPath)
+
+	for i, inst := range instances {
+		if i == 0 {
+			inst.shareLeader(t, n, n, 2, beaconID, sch)
+		} else {
+			inst.share(t, instances[0].addr, beaconID)
+		}
+		time.Sleep(500 * time.Millisecond)
+	}
+
+	defer func() {
+		for _, inst := range instances {
+			inst.stopAll()
+		}
+	}()
+
+	time.Sleep(1 * time.Second)
+
+	// try to reload a beacon which is already loaded
+	err := instances[3].reload(beaconID)
+	require.Error(t, err)
+
+	// wait some time to generate some randomness
+	time.Sleep(1 * time.Minute)
+
+	// Stop beacon process... not the entire node
+	err = instances[3].stop(beaconID)
+	require.NoError(t, err)
+
+	// check the node is still alive
+	testPing(t, instances[3].ctrlPort)
+
+	// reload a beacon
+	err = instances[3].reload(beaconID)
+	require.NoError(t, err)
```

Comment:
how much harder would it be to have the stopped node prevent generation of a beacon, and confirm that no beacon is generated while the node is stopped, but then resume after it is reloaded?

Reply:
I can add that test too! I will do it in another PR.